pages/api/text.ts

```diff
@@ -0,0 +1,62 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql, OperationResult } from 'urql'
+import { fetchData, getEns } from './_utils'
+
+const ProfileTextRecordsQuery = gql<{
+  domains: [{ resolver: { texts: string[] } }]
+}>`
+  query ProfileTextRecords($name: String!) {
+    domains(where: { name: $name }) {
+      resolver {
+        texts
+      }
+    }
+  }
+`
+
+export async function getEnsTextRecords(
+  ensName: string
+): Promise<{ key: string; value: string }[]> {
+  // 1. Check which text records are set for the domain with ENS subgraph,
```

Comment:
Right, I remember. That's actually bad as the whole flow of getting text records is made dependent on this call but most likely there is no alternative yet until new ens.js is done if we want to save contract calls. Basically problem is that we need to call all text record fields to figure out if it has been set or not.

We could kick out the initial subgraph call and see how much stress this puts on our infura

Reply:
This might apply here too so it could be this would even stop working in October https://twitter.com/Mardeni01/status/1560285524920860672

Reply:
I've removed the subgraph call for now but I can put it back if it places to many requests on Infura